service/api/service.go

```diff
@@ -143,17 +141,46 @@ func (s *service) AddRecord(ctx context.Context, req *pb.AddRecordRequest) (*pb.
 	if err != nil {
 		return nil, err
 	}
-	rec, err := s.s.AddRecord(ctx, threadID, body)
+	rec, err := s.s.CreateRecord(ctx, threadID, body)
+	if err != nil {
+		return nil, err
+	}
+	prec, err := cbor.RecordToProto(ctx, s.s, rec.Value())
 	if err != nil {
 		return nil, err
 	}
 	return &pb.NewRecordReply{
 		ThreadID: rec.ThreadID().Bytes(),
 		LogID:    marshalPeerID(rec.LogID()),
-		Record:   pbRecordFromRecord(rec.Value()),
+		Record:   util.RecFromServiceRec(prec),
 	}, nil
 }
 
+func (s *service) AddRecord(ctx context.Context, req *pb.AddRecordRequest) (*pb.AddRecordReply, error) {
+	log.Debugf("received add record request")
+
+	threadID, err := thread.Cast(req.ThreadID)
+	if err != nil {
+		return nil, err
+	}
+	info, err := s.s.GetThread(ctx, threadID)
+	if err != nil {
+		return nil, err
+	}
+	logID, err := peer.IDFromBytes(req.LogID)
```

Comment:
The client tells us what logID to add the record to. The record will have a valid signature if they have "permission" to do this.